src/bsoncxx/include/bsoncxx/v1/detail/prelude.hpp

```diff
@@ -0,0 +1,118 @@
+// Copyright 2009-present MongoDB, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma push_macro("BSONCXX_PRIVATE_V1_INSIDE_MACRO_GUARD_SCOPE")
+#undef BSONCXX_PRIVATE_V1_INSIDE_MACRO_GUARD_SCOPE
+#define BSONCXX_PRIVATE_V1_INSIDE_MACRO_GUARD_SCOPE
```

Comment:
A dedicated scope macro can be used to track and enforce correct 1-to-1 inclusion of the prelude and postlude headers fron their initiation, unlike with v_noabi macro guard headers (CXX-2770 + continued backward compatibility with the v_noabi include-via-prelude pattern).